deepinv/sampling/samplers.py

```diff
@@ -0,0 +1,347 @@
+        return self.sample(y, physics, X_init=X_init, seed=seed)[0]
+
+    def sample(
+        self,
+        y: torch.Tensor,
+        physics: Physics,
+        X_init: Union[torch.Tensor, None] = None,
+        seed: Union[int, None] = None,
+        g_statistics=[lambda x: x],
+        **kwargs,
+    ) -> tuple[torch.Tensor, torch.Tensor]:
+        r"""
+        Execute the MCMC sampling chain and compute posterior statistics.
+
+        This method runs the main MCMC sampling loop to generate samples from the posterior
+        distribution and compute their statistics using Welford's online algorithm.
+
+        :param torch.Tensor y: The observed measurements/data tensor
+        :param Physics physics: Forward operator of your inverse problem.
+        :param torch.Tensor X_init: Initial state of the Markov chain. If None, uses ``physics.A_adjoint(y)`` as the starting point
+            Default: ``None``
+        :param int seed: Optional random seed for reproducible sampling.
+            Default: ``None``
+        :param list g_statistics: List of functions for which to compute posterior statistics. Default: ``[lambda x: x]``
+        :param kwargs: Additional arguments passed to the sampling iterator (e.g., proposal distributions)
+        :return: | If a single g_statistic was specified: Returns tuple (mean, var) of torch.Tensors
+            | If multiple g_statistics were specified: Returns tuple (means, vars) of lists of torch.Tensors
+
+        Example:
+            >>> # Basic usage with default settings
+            >>> sampler = BaseSample(iterator, data_fidelity, prior)
+            >>> mean, var = sampler(measurements, forward_operator)
+
+            >>> # Using multiple statistics
+            >>> sampler = BaseSample(
+            ...     iterator, data_fidelity, prior,
+            ...     g_statistics=[lambda x: x, lambda x: x**2]
+            ... )
+            >>> means, vars = sampler(measurements, forward_operator)
+        """
+        # Set random seed if provided
+        if seed is not None:
+            torch.manual_seed(seed)
+
+        # Initialization
+        if X_init is None:
+            X_t = physics.A_adjoint(y)
+        else:
+            X_t = X_init
+
+        if self.history:
+            if isinstance(self.history, deque):
+                self.history = deque([X_t], maxlen=self.history_size)
+            else:
+                self.history = [X_t]
+
+        self.mean_convergence = False
+        self.var_convergence = False
+
+        # Initialize Welford trackers for each g_statistic
+        statistics = []
+        for g in g_statistics:
+            statistics.append(Welford(g(X_t)))
+
+        # Initialize for convergence checking
+        mean_prevs = [stat.mean().clone() for stat in statistics]
+        var_prevs = [stat.var().clone() for stat in statistics]
+
+        # Run the chain
+        for i in tqdm(range(self.max_iter), disable=(not self.verbose)):
```

Comment:
It seems more clear to me if we use `it` rather than `i` for the iterator variable

Reply:
.